ui/perfherder/compare/CompareSubtestsView.jsx

```diff
@@ -0,0 +1,253 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { react2angular } from 'react2angular/index.es2015';
+import difference from 'lodash/difference';
+
+import perf from '../../js/perf';
+import { createQueryParams } from '../../helpers/url';
+import {
+  createNoiseMetric,
+  getCounterMap,
+  createGraphsLinks,
+} from '../helpers';
+import { noiseMetricTitle } from '../constants';
+
+import withValidation from './Validation';
+import CompareTableView from './CompareTableView';
+
+// TODO remove $stateParams and $state after switching to react router
+export class CompareSubtestsView extends React.PureComponent {
+  createQueryParams = (parent_signature, repository, framework) => ({
+    parent_signature,
+    framework,
+    repository,
+  });
+
+  getQueryParams = (timeRange, framework) => {
+    const {
+      originalProject,
+      newProject,
+      originalRevision,
+      newRevision,
+      newResultSet,
+      originalSignature,
+      newSignature,
+    } = this.props.validated;
+
+    const originalParams = this.createQueryParams(
+      originalSignature,
+      originalProject,
+      framework.id,
+    );
+
+    if (originalRevision) {
+      originalParams.revision = originalRevision;
+    } else {
+      // can create a helper function for both views
+      const startDateMs =
+        (newResultSet.push_timestamp - timeRange.value) * 1000;
+      const endDateMs = newResultSet.push_timestamp * 1000;
+      originalParams.startday = new Date(startDateMs)
+        .toISOString()
+        .slice(0, -5);
+      originalParams.endday = new Date(endDateMs).toISOString().slice(0, -5);
+    }
+
+    const newParams = this.createQueryParams(
+      newSignature,
+      newProject,
+      framework.id,
+    );
+    newParams.revision = newRevision;
+    return [originalParams, newParams];
+  };
+
+  createLinks = (oldResults, newResults, timeRange, framework) => {
+    const {
+      originalProject,
+      newProject,
+      originalRevision,
+      newRevision,
+    } = this.props.validated;
+    let links = [];
+
+    if (framework.name === 'talos' && originalRevision) {
```

Comment:
I think when @SJasoria wrote up the distribution viewer, we only had the required perfherder-data.json structure for talos. But it appears now we have the same data for other jobs, e.g. raptor:

https://treeherder.mozilla.org/#/jobs?repo=mozilla-inbound&revision=70c2025fdad9077ac5021fc6514aa9466573d3bc&searchStr=raptor&selectedJob=235421758

It might be worth considering relaxing this restriction then, either here or in a followup.

Reply:
Oh, good point. Thanks for mentioning it.